interfaces/apparmor/spec.go

```diff
@@ -260,8 +260,64 @@ func chopTree(path string, assumedPrefixDepth int) (left, right []string, err er
 	return left, right, nil
 }
 
+// WritableMimicProfile writes apparmor rules for a writable mimic at the given path.
+func WritableMimicProfile(buf *bytes.Buffer, path string, assumedPrefixDepth int) {
+	fmt.Fprintf(buf, "  # Writable mimic %s\n", path)
+
+	iter, err := strutil.NewPathIterator(path)
+	if err != nil {
+		panic(err)
+	}
+
+	// Handle the prefix that is assumed to exist first.
+	fmt.Fprintf(buf, "  # .. permissions for traversing the prefix that is assumed to exist\n")
+	for iter.Next() {
+		if iter.Depth() < assumedPrefixDepth {
+			fmt.Fprintf(buf, "  %s r,\n", iter.CurrentPath())
+		}
+	}
+
+	// Rewind the iterator and handle the part that needs to be created.
+	iter.Rewind()
+	for iter.Next() {
+		if iter.Depth() >= assumedPrefixDepth {
```

Comment:
It's more clear if the logic is inverted, and everything else is dedented:
```
if iter.Depth() < assumedPrefixDepth {
    continue
}
```
otherwise we need to read til the end to figure that there's nothing else after this branch.